File: fizzbuzz.test.js
const fizzBuzz = require('./fizzbuzz');

describe('FizzBuzz function', () => {
  test.each([
    [1, '1'],
    [2, '2'],
    [3, 'Fizz'],
    [4, '4'],
    [5, 'Buzz'],
    [6, 'Fizz'],
    [10, 'Buzz'],
    [15, 'FizzBuzz'],
    [30, 'FizzBuzz'],
    [97, '97'],
    [100, 'Buzz']
  ])('fizzBuzz(%i) should return %s', (input, expected) => {
    expect(fizzBuzz(input)).toBe(expected);
  });

  test('fizzBuzz returns correct values for numbers 1 to 100', () => {
    for (let i = 1; i <= 100; i++) {
      let expected = '';
      if (i % 3 === 0) expected += 'Fizz';
      if (i % 5 === 0) expected += 'Buzz';
      if (expected === '') expected = i.toString();
      expect(fizzBuzz(i)).toBe(expected);
    }
  });
});

File: tests/fizzbuzz_test.py
"""
Unit tests for the FizzBuzz problem.

These tests verify the correctness of a FizzBuzz implementation.

Rules tested:
- Numbers divisible by 3 return "Fizz"
- Numbers divisible by 5 return "Buzz"
- Numbers divisible by both 3 and 5 return "FizzBuzz"
- Other numbers return their string representation

Usage:
- Implement a function `fizzbuzz(n: int) -> str` in your code.
- Run these tests with pytest to validate your implementation.

Note: The FizzBuzz function is not implemented here.
"""
import pytest

# The tests expect a function fizzbuzz(n) to be implemented by the user.
# To run tests, implement fizzbuzz(n) in your codebase.

@pytest.mark.parametrize("input_value,expected_output", [
    (1, "1"),
    (2, "2"),
    (3, "Fizz"),
    (4, "4"),
    (5, "Buzz"),
    (6, "Fizz"),
    (7, "7"),
    (8, "8"),
    (9, "Fizz"),
    (10, "Buzz"),
    (11, "11"),
    (12, "Fizz"),
    (13, "13"),
    (14, "14"),
    (15, "FizzBuzz"),
])
def test_fizzbuzz(input_value, expected_output):
    """Test fizzbuzz output for given input values."""
    # Import fizzbuzz function here to allow user implementation
    from fizzbuzz import fizzbuzz
    assert fizzbuzz(input_value) == expected_output
